apache-poi-3/src/test/java/com/baeldung/poi/rowstyle/PoiBoldStyleIntegrationTest.java

```diff
@@ -0,0 +1,152 @@
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.streaming.SXSSFWorkbook;
+import org.apache.poi.xssf.usermodel.XSSFCellStyle;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.junit.jupiter.api.Test;
+
+public class PoiBoldStyleIntegrationTest {
+
+    private void writeSampleSheet(Path destination, Workbook workbook) throws FileNotFoundException, IOException {
+        Sheet sheet = workbook.createSheet();
+        CellStyle boldStyle = PoiUtils.boldFontStyle(workbook);
+
+        Row header = PoiUtils.newRow(sheet, "Name", "Value", "Details");
+        header.setRowStyle(boldStyle);
+
+        PoiUtils.newRow(sheet, "Albert", "A", "First");
+        PoiUtils.newRow(sheet, "Jane", "B", "Second");
+        PoiUtils.newRow(sheet, "Zack", "C", "Third");
+
+        PoiUtils.write(workbook, destination);
+    }
+
+    private void assertRowStyleAppliedAndDefaultCellStylesDontMatch(Path sheetFile) throws IOException, InvalidFormatException {
+        try (Workbook workbook = new XSSFWorkbook(sheetFile.toFile())) {
+            Sheet sheet = workbook.getSheetAt(0);
+            Row row0 = sheet.getRow(0);
+
+            final XSSFCellStyle rowStyle = (XSSFCellStyle) row0.getRowStyle();
+            assertTrue(rowStyle.getFont()
+                .getBold());
+
+            row0.forEach(cell -> {
+                XSSFCellStyle style = (XSSFCellStyle) cell.getCellStyle();
+                assertNotEquals(rowStyle, style);
+            });
+
+            Row row1 = sheet.getRow(1);
+            XSSFCellStyle row1Style = (XSSFCellStyle) row1.getRowStyle();
+            assertNull(row1Style);
+
+            Files.delete(sheetFile);
+        }
+    }
+
+    @Test
+    void givenXssfWorkbook_whenSetRowStyle1stRow_thenOnly1stRowStyled() throws IOException, InvalidFormatException {
+        Path sheetFile = Files.createTempFile("xssf-row-style", ".xlsx");
+
+        try (Workbook workbook = new XSSFWorkbook()) {
+            writeSampleSheet(sheetFile, workbook);
+        }
+
+        assertRowStyleAppliedAndDefaultCellStylesDontMatch(sheetFile);
+    }
+
+    @Test
+    void givenSxssfWorkbook_whenSetRowStyle_thenOnly1stRowStyled() throws IOException, InvalidFormatException {
+        Path sheetFile = Files.createTempFile("sxssf-row-style", ".xlsx");
+
+        try (Workbook workbook = new SXSSFWorkbook()) {
+            writeSampleSheet(sheetFile, workbook);
+        }
+
+        assertRowStyleAppliedAndDefaultCellStylesDontMatch(sheetFile);
+    }
+
+    @Test
+    void givenHssfWorkbook_whenSetRowStyle_thenOnly1stRowStyled() throws IOException {
+        Path sheetFile = Files.createTempFile("hssf-row-style", ".xls");
+
+        try (Workbook workbook = new HSSFWorkbook()) {
+            writeSampleSheet(sheetFile, workbook);
+        }
+
+        try (Workbook workbook = new HSSFWorkbook(new FileInputStream(sheetFile.toFile()))) {
```

Comment:
Using Files.newInputStream is generally preferred over new FileInputStream.

```suggestion
        try (Workbook workbook = new HSSFWorkbook(new FileInputStream(sheetFile.toFile()))) {
```